Backends/CLX/port.lisp

```diff
@@ -214,33 +214,23 @@
                       :map (sheet-enabled-p sheet)))
 
 (defmethod %realize-mirror ((port clx-port) (sheet top-level-sheet-pane))
-  (let ((q (compose-space sheet))
-        (frame (pane-frame sheet)))
-    (let ((window (realize-mirror-aux
-                   port sheet
-                   :map nil
-                   :width (round-coordinate (space-requirement-width q))
-                   :height (round-coordinate (space-requirement-height q))))
-	  (name (frame-pretty-name frame)))
-      (setf (xlib:wm-hints window) (xlib:make-wm-hints :input :on))
-      (setf (xlib:wm-name window) name)
-      (xlib:change-property window
-                            :_NET_WM_NAME
-                            (babel:string-to-octets name :encoding :utf-8)
-                            :UTF8_STRING 8)
-      (setf (xlib:wm-icon-name window) name)
-      (xlib:change-property window
-                            :_NET_WM_ICON_NAME
-                            (babel:string-to-octets name :encoding :utf-8)
-                            :UTF8_STRING 8)
-      (xlib:set-wm-class
-       window
-       (string-downcase (frame-name frame))
-       (string-capitalize (frame-name frame)))
-      (setf (xlib:wm-protocols window) `(:wm_delete_window))
-      (xlib:change-property window
-                            :WM_CLIENT_LEADER (list (xlib:window-id window))
-                            :WINDOW 32))))
+  (let* ((q (compose-space sheet))
+         (frame (pane-frame sheet))
+         (window (realize-mirror-aux
+                  port sheet
+                  :map nil
+                  :width (round-coordinate (space-requirement-width q))
+                  :height (round-coordinate (space-requirement-height q))))
+         (name (frame-name frame))
+         (pretty-name (frame-pretty-name frame)))
+    (setf (xlib:wm-hints window) (xlib:make-wm-hints :input :on))
+    (%set-window-name window pretty-name)
+    (%set-window-icon-name window pretty-name)
+    (xlib:set-wm-class window (string-downcase name) (string-capitalize name))
+    (setf (xlib:wm-protocols window) `(:wm_delete_window))
+    (xlib:change-property window
+                          :WM_CLIENT_LEADER (list (xlib:window-id window))
+                          :WINDOW 32)))
```

Comment:
frame references could be removed from here if we had created `:after` method on `realize-mirror` in `panes.lisp` which calls the name setter. I'm not sure what to do with wm-class though in that case.

Reply:
The sheet would need a name and a "pretty name" to make that possible. Are those in scope for the sheet protocol extension you suggested above?

Reply:
can't it have only name? is it because pretty name is a string?

I imagine, that method on top-level sheet would setf this pane's name to the frame's pretty name, I was thinking about sheets name as a string designator. sheets are considered in clim abstraction as windows, I don't think there is a necessity to have two names for them.

Reply:
Having a string as a name for the sheet would makes, yes. But currently `%realize-mirror` also uses the `frame-name` (i.e. not pretty) for the window class. If it hadn't access to the frame, and the sheet also didn't have this second name, how would that work?

Reply:
I know that it works like that right now, but is there a reason to not set wm-class to the same thing?

Reply:
I'm certainly not an expert regarding ICCCM, but it sounds like the concepts are actually different.

The two `WM_CLASS` strings 
* identify the application instance and class (not e.g. a document or state within the window) for resource lookup and i presume automated window management
* are typically set at startup and not changed (in fact, as i understand it, they are not allowed to change while the window is visible)

The `_NET_WM_NAME` property
* should be displayed as the title of the window (other applications typically include things like current document title or status)
* can, as far as i can tell, change at any time (other applications do this. for example, gedit changes this property when a document is selected)

So merging the two doesn't seem right. In addition, the `frame-name`/`frame-pretty-name` pair seems to match the `WM_CLASS`/`_NET_WM_NAME` pair pretty well, given the description in the CLIM spec.

That said, I already agreed that having this work for sheets without relying on an associated frame would be good. My conclusion is that the sheet has to mirror the name/pretty name distinction. And why shouldn't it? For the IMGUI case, this separation would allow IMGUI windows to work with resources and window managers in the normal way while being able to control and change the window title independently.

Reply:
sounds reasonable